### PR TITLE
FortiManager Web Filter Security Profile

### PR DESCRIPTION
**Pull request by ftntcorecse**
##### SUMMARY
Managing web filter security profiles for FortiManager

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

fmgr_secprof_web

##### ANSIBLE VERSION

```paste below
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cse/PycharmProjects/JPMC_FOS/ansible/lib/ansible/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
``

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

### CODE REVIEW

**Comment by ftntcorecse** [comment]
Hi All,

May I have a status update for this PR?  Is anything else needed from me to move this forward?

Thanks

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py [hunk 1]
```diff
@@ -0,0 +1,1236 @@
...
+    description:
+      - Sets one of three modes for managing the object.
+      - Allows use of soft-adds instead of overwriting existing values
+    choices: ['add', 'set', 'delete', 'update']
+    required: false
+    default: add
+
+  youtube_channel_status:
+    description:
+      - YouTube channel filter status.
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py:70]: Please format and append description in one single statement wherever feasible

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py [hunk 2]
```diff
@@ -0,0 +1,1236 @@
...
+    required: false
+    choices: ["session", "timeout"]
+
+  ftgd_wf_filters_warning_prompt:
+    description:
+      - Warning prompts in each category or each domain.
+      - choice | per-domain | Per-domain warnings.
+      - choice | per-category | Per-category warnings.
+    required: false
+    choices: ["per-domain", "per-category"]
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py:447]: To be consistent with other Ansible modules kindly modify the choices options at all required places as:
```suggestion
    choices: 
    - per-domain
    - per-category
```

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py [hunk 3]
```diff
@@ -0,0 +1,1236 @@
...
+
+
+###############
+# START METHODS
+###############
+
+
+def fmgr_webfilter_profile_addsetdelete(fmg, paramgram):
+    """
+    fmgr_webfilter_profile -- Your Description here, bruh
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py]: Please update and remove the generic comment.

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py [hunk 4]
```diff
@@ -0,0 +1,1236 @@
...
+    description:
+      - The username associated with the account.
+    required: true
+
+  password:
+    description:
+      - The password associated with the username account.
+    required: true
+
+  mode:
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py:60]: if `mode` is similar to `state`, please change the same to `state` to be in consistent with other Ansible modules

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py (hunk 1 above)
Comment (ftntcorecse) [reply lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py:70]: Hi @justjais For our module we need to have comments for every available choice.  What do you recommend we do for multiple choices each needing a description?  Thank you for your help

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py (hunk 4 above)
Comment (ftntcorecse) [reply lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py:60]: Hi @justjais We use mode to be different than state, mode in our modules is the HTTP mode used for the API call.  Thanks

**Comment by ftntcorecse** [comment]
Hi @justjais I've added fixes to your comments above.  Thanks again

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py [hunk 5]
```diff
@@ -28,9 +28,9 @@
 module: fmgr_secprof_web
 version_added: "2.8"
 author:
-    - Luke Weighall (@lweighall)
-    - Andrew Welsh (@Ghilli3)
-    - Jim Huber (@p4r4n0y1ng)
+    - Luke Weighall (lweighall)
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_web.py]: @ftntcorecse was there any specific reason for removing the `@` as ansible modules follow author with `@` sign.

**Comment by mattclay** [comment]
Closing and re-opening to trigger CI.

**Review by justjais — APPROVED** [review]

**Pull request merged**